src/components/ScmRepositoryDetailPage/ScmRepositoryDetailPage.tsx

```diff
@@ -0,0 +1,132 @@
+import { Button, ClipboardCopy, Text, TextContent, TextVariants } from '@patternfly/react-core';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+import { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { useServiceContainer } from 'hooks/useServiceContainer';
+import { useTitle } from 'hooks/useTitle';
+
+import { ActionButton } from 'components/ActionButton/ActionButton';
+import { AttributesItems } from 'components/AttributesItems/AttributesItems';
+import { ContentBox } from 'components/ContentBox/ContentBox';
+import { CopyToClipboard } from 'components/CopyToClipboard/CopyToClipboard';
+import { PageLayout } from 'components/PageLayout/PageLayout';
+import { ServiceContainerLoading } from 'components/ServiceContainers/ServiceContainerLoading';
+import { Toolbar } from 'components/Toolbar/Toolbar';
+import { ToolbarItem } from 'components/Toolbar/ToolbarItem';
+import { TooltipText } from 'components/TooltipText/TooltipText';
+
+import * as scmRepositoryApi from 'services/scmRepositoryApi';
+
+import { generatePageTitle } from 'utils/titleHelper';
+
+export const ScmRepositoryDetailPage = () => {
+  const { scmRepositoryId } = useParams();
+
+  const serviceContainerScmRepository = useServiceContainer(scmRepositoryApi.getScmRepository);
+  const serviceContainerScmRepositoryRunner = serviceContainerScmRepository.run;
+
+  useEffect(() => {
+    serviceContainerScmRepositoryRunner({ serviceData: { id: scmRepositoryId } });
+  }, [serviceContainerScmRepositoryRunner, scmRepositoryId]);
+
+  useTitle(
+    generatePageTitle({
+      serviceContainer: serviceContainerScmRepository,
+      entityName: serviceContainerScmRepository.data?.id,
+      firstLevelEntity: 'SCM Repository',
+    })
+  );
+
+  const InternalScmRepositoryLink = (internalUrl: string) => {
+    /**
+     * Parses internal repo url to Gerrit gitweb link of the project
+     */
+    const parseInternalRepoLink = (url: string) => {
+      const protocol = url.split('://')[0];
+      const base = url.split('://')[1].split('/')[0];
+      const project = url.split(base + (['https', 'http'].includes(protocol) ? '/gerrit/' : '/'))[1];
+      return 'https://' + base + '/gerrit/gitweb?p=' + project + ';a=summary';
+    };
+    return (
+      <CopyToClipboard
+        suffixComponent={
+          <Button
+            component="a"
+            href={parseInternalRepoLink(internalUrl)}
+            target="_blank"
+            rel="noopener noreferrer"
+            variant="tertiary"
+            icon={<ExternalLinkAltIcon />}
+          >
+            Gerrit
+          </Button>
+        }
+      >
+        {parseInternalRepoLink(internalUrl)}
+      </CopyToClipboard>
+    );
+  };
+
+  const parseScmRepositoryTitle = (internalUrl: string) => (internalUrl ? internalUrl.split('/').splice(3).join('/') : '');
```

Comment:
Move it into global scope, above this component. Should it be useful in the future elsewhere, move it somewhere else, maybe `utils.ts`?